honeybee_schema/_openapi.py

```diff
@@ -81,6 +81,25 @@ def get_openapi(
         }
         tags.append(tag)
 
+        # sort properties order: put required parameters at begining of the list
+        s = schemas[name]
+        if 'required' in s:
+            props = s['properties']
+            required = s['required']
+
+            propKeys = props.keys()
+            sortedProps = {}
+            optional = {}
+            for r in propKeys:
+                if r in required:
+                    sortedProps[r] = props[r]
+                else:
+                    optional[r] = props[r]
+
+            sortedProps.update(optional)
+
+            s['properties'] = sortedProps
+
```

Comment:
I like this addition but I suggest to make some small changes to the code:

```python
        s = schemas[name]
        if not 'required' in s:
            continue
        properties = s['properties']
        required = s['required']

        sorted_props = {}
        optional = {}
        for prop, value in properties.items():
            if prop in required:
                sorted_props[prop] = value
            else:
                optional[prop] = value

        sorted_props.update(optional)

        s['properties'] = sorted_props
```

Reply:
Ha, thanks @mostaphaRoudsari , I remember there is something like "items()" for object, but did found it is items(). Thanks, now looks much better!